utils/cloudinfo/cloudinfo.go

```diff
@@ -23,19 +23,23 @@ import (
 type CloudInfo interface {
 	GetCloudProvider() info.CloudProvider
 	GetInstanceType() info.InstanceType
+	GetInstanceID() info.InstanceID
 }
 
 type realCloudInfo struct {
 	cloudProvider info.CloudProvider
 	instanceType  info.InstanceType
+	instanceID    info.InstanceID
 }
 
 func NewRealCloudInfo() CloudInfo {
 	cloudProvider := detectCloudProvider()
 	instanceType := detectInstanceType(cloudProvider)
+	instanceID := detectInstanceID(cloudProvider)
```

Comment:
I think detectCloudProvider could take some time. The others will be almost immediate as we are accessing the correct API server. In short this function could block for some time and its users should be aware of this as it was already blocking before this patch.